src/orange/orangeinp/UnitProto.cc

```diff
@@ -185,11 +182,11 @@ void UnitProto::build(ProtoBuilder& input) const
         VolumeInput vi;
 
         // Construct logic and faces with remapped surfaces
-        auto&& [faces, logic]
-            = build_logic.operator()<detail::PostfixLogicBuilderPolicy>(
-                node_id);
+        auto faces = build_logic(
+            detail::PostfixLogicBuilderPolicy{
+                csg_unit.tree, &sorted_local_surfaces, &vi.logic},
+            node_id);
```

Comment:
I think this still needs another iteration... getting one return result by reference and another by return value is inconsistent/surprising.

Reply:
42bd6c0 and 32a9fdd are the two alternatives. I think that returning by value and moving from lgc is nicer to use.

Reply:
Definitely agree: returning by value (32a9) is better.